### ISSUE TITLE
template string error

### ISSUE THREAD

**Issue by backspaces** [decaffeinate/decaffeinate#212]
I'm converting a fairly large coffeescript repo to es6, and loving decaffeinate! This snippet from a larger coffee module produces scrambled template strings.  This:
```coffee
Color = {
  rgbaString: (r, g, b, a=255) ->
    a = a/255; a4 = a.toPrecision(4)
    if a is 1 then "rgb(#{r},#{g},#{b})" else "rgba(#{r},#{g},#{b},#{a4})"
  hslString: (h, s, l, a=255) ->
    a = a/255; a4 = a.toPrecision(4)
    if a is 1 then "hsl(#{h},#{s}%,#{l}%)" else "hsla(#{h},#{s}%,#{l}%,#{a4})"
}
```
Produces this; using decaffeinate@2.9.1: the `$gb(#{r}` should be `rgb(${r}` etc.
```js
let Color = {
  rgbaString(r, g, b, a=255) {
    a = a/255; let a4 = a.toPrecision(4);
    if (a === 1) { return `$gb(#{r},${g},${b})`; } else { return `$gba(#{r},${g},${b},${a4})`; }
  },
  hslString(h, s, l, a=255) {
    a = a/255; let a4 = a.toPrecision(4);
    if (a === 1) { return `$sl(#{h},${s}%,${l}%)`; } else { return `$sla(#{h},${s}%,${l}%,${a4})`; }
  }
};
```

BTW: I use this bashrc one-liner to find this sort of thing:
`clipcaff (){ pbpaste | decaffeinate;}`
and esling --fix is great for converting the decaffeinate code into my .eslintrc style!

**Comment by backspaces** [comment]
Oops, forgot to include the coffeescript generated code, in case that helps.
```
// Generated by CoffeeScript 1.10.0
(function() {
  var Color;

  Color = {
    rgbaString: function(r, g, b, a) {
      var a4;
      if (a == null) {
        a = 255;
      }
      a = a / 255;
      a4 = a.toPrecision(4);
      if (a === 1) {
        return "rgb(" + r + "," + g + "," + b + ")";
      } else {
        return "rgba(" + r + "," + g + "," + b + "," + a4 + ")";
      }
    },
    hslString: function(h, s, l, a) {
      var a4;
      if (a == null) {
        a = 255;
      }
      a = a / 255;
      a4 = a.toPrecision(4);
      if (a === 1) {
        return "hsl(" + h + "," + s + "%," + l + "%)";
      } else {
        return "hsla(" + h + "," + s + "%," + l + "%," + a4 + ")";
      }
    }
  };

}).call(this);
```

**Comment by thebanjomatic** [comment]
I ran into this with my own project the other day. As a post conversion step, I have  been searching the converted files for `#{` to fix things up manually.

**Comment by eventualbuddha** [comment]
Weird. [This](http://decaffeinate.github.io/decaffeinate/repl/#?evaluate=true&code=%22(%23%7Br%7D%22%0A) is the smallest reproducible test case I could come up with:

```coffee
"(#{r}"
```

The parenthesis is messing it up somehow.

**Comment by Eden-Sun** [comment]
parentheses issue on literal
https://github.com/decaffeinate/decaffeinate-parser/pull/12 
is that help?

**Comment by Eden-Sun** [comment]
It seems many problems to combine *coffee-compiled* strings back to an origin *es6-like* literal.
I am expecting to see your solution.

**Comment by eventualbuddha** [comment]
Fixed in v2.11.1 ([repl](http://decaffeinate.github.io/decaffeinate/repl/#?evaluate=true&code=Color%20%3D%20%7B%0A%20%20rgbaString%3A%20(r%2C%20g%2C%20b%2C%20a%3D255)%20-%3E%0A%20%20%20%20a%20%3D%20a%2F255%3B%20a4%20%3D%20a.toPrecision(4)%0A%20%20%20%20if%20a%20is%201%20then%20%22rgb(%23%7Br%7D%2C%23%7Bg%7D%2C%23%7Bb%7D)%22%20else%20%22rgba(%23%7Br%7D%2C%23%7Bg%7D%2C%23%7Bb%7D%2C%23%7Ba4%7D)%22%0A%20%20hslString%3A%20(h%2C%20s%2C%20l%2C%20a%3D255)%20-%3E%0A%20%20%20%20a%20%3D%20a%2F255%3B%20a4%20%3D%20a.toPrecision(4)%0A%20%20%20%20if%20a%20is%201%20then%20%22hsl(%23%7Bh%7D%2C%23%7Bs%7D%25%2C%23%7Bl%7D%25)%22%20else%20%22hsla(%23%7Bh%7D%2C%23%7Bs%7D%25%2C%23%7Bl%7D%25%2C%23%7Ba4%7D)%22%0A%7D)).
